src/pages/index.js

```diff
@@ -118,75 +167,129 @@ function getCardElement(data) {
   return cardElement;
 }
 
-function handleEscape(evt) {
-  if (evt.key === "Escape") {
-    const openedModal = document.querySelector(".modal_is-opened");
-    if (openedModal) {
-      closeModal(openedModal);
-    }
-  }
-}
+// Delete form submit handler
+deleteModalForm.addEventListener("submit", function handleDeleteSubmit(evt) {
+  evt.preventDefault();
+  deleteBtnText(deleteModalForm, true, "delete", "deleting...");
+  if (!selectedCardId || !selectedCard) return;
+
+  api
 
-profileEditBtn.addEventListener("click", function () {
+    .deleteCard(selectedCardId)
+    .then(() => {
+      selectedCard.remove();
+      closeModal(deleteModal);
+    })
+
+    .catch(console.error)
+    .finally(() => {
+      selectedCard = null;
+      selectedCardId = null;
+    });
+});
+
+// Modal close handlers for delete modal
+deleteModalCloseBtn.addEventListener("click", () => closeModal(deleteModal));
+deleteCancelBtn.addEventListener("click", () => closeModal(deleteModal));
+
+// Profile edit handlers
+profileEditBtn.addEventListener("click", () => {
   editProfileNameInput.value = profileNameEl.textContent;
   editProfileDescriptionInput.value = profileDescriptionEl.textContent;
-  openModal(editProfileModal);
   resetValidation(
     editProfileForm,
     [editProfileNameInput, editProfileDescriptionInput],
     settings
   );
+  openModal(editProfileModal);
 });
-
-profileEditCloseBtn.addEventListener("click", function () {
-  closeModal(editProfileModal);
-});
-
-previewModalCloseBtn.addEventListener("click", function () {
-  closeModal(previewModal);
-});
-
-newPostBtn.addEventListener("click", function () {
-  openModal(newPostCardModal);
-});
-
-newPostCloseBtn.addEventListener("click", function () {
-  closeModal(newPostCardModal);
-});
-
-function handleEditProfileSubmit(evt) {
-  evt.preventDefault();
-  profileNameEl.textContent = editProfileNameInput.value;
-  profileDescriptionEl.textContent = editProfileDescriptionInput.value;
-  closeModal(editProfileModal);
-}
-editProfileForm.addEventListener("submit", handleEditProfileSubmit);
-
-function handleAddCardSubmit(evt) {
+profileEditCloseBtn.addEventListener("click", () =>
+  closeModal(editProfileModal)
+);
+editProfileForm.addEventListener(
+  "submit",
+  function handleEditProfileSubmit(evt) {
+    evt.preventDefault();
+    //submitBtn.textContent = "saving..."
+    setButtonText(submitBtn, true, "save", "saving...");
```

Comment:
Please, use the correct submit buttons in every request. And implement the `setButtonText()` function correctly